code/game/mob/living/carbon/human/human_defense.dm

```diff
@@ -611,8 +611,12 @@ bullet_act
 	if (hit_zone == "l_hand" || hit_zone == "r_hand" || hit_zone == "r_foot" || hit_zone == "l_foot")
 		if (prob(50))
 			if (prob(65))
-				visible_message("<span class='notice'>[user] has tried to strike [src]'s [affecting.name] with [I.name] but missed!</span>")
-				return
+				if (user != src)
```

Comment:
Same here, no need to make it more complex

Reply:
No, this checks if the person is attacking themselves instead of [src], it's meant to be from different views of the clients to make it more immersive.